### ISSUE TITLE
Button rendering issue in Autocombat UI with localized buttons

### ISSUE THREAD

**Issue by felix-schneider** [ihhub/fheroes2#9261]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

There is an ugly border around the "restart" button in the German localization. This seems to be independent of the resolution, I tried multiple.

![image](https://github.com/user-attachments/assets/462beb2c-677b-4e95-a6ff-eda58ab26145)

### Save file

[Alteris_0001.zip](https://github.com/user-attachments/files/17661168/Alteris_0001.zip)

**Comment by Branikolog** [comment]
There's a similar problem for Russian resources for a pressed "Cancel" button for campaign window:

![Image](https://github.com/user-attachments/assets/9a4f592c-fc0e-44cf-a150-8ef6e152edb1)

**Comment by Branikolog** [comment]
Hi, @zenseii 
I believe, we can switch to engine generated buttons in 98% of places when speaking about the Russian version.
Localized buttons have also a bunch of issues (most common is wrong text shifting for the pressed state).

>Meanwhile for Russian I don't have these assets available so I don't know what remains there until we do a complete replacement.

I can remember just the battle skip and auto and the editor specs button among those, which need translation. They all have mentioned shift issue, but not in such a noticeable way other buttons have. So, It's ok to leave them as it is for now and change all other buttons for fheroes2 new ones.

**Comment by Branikolog** [comment]
@zenseii 

>We could perhaps make a shield similar to the one in the upper right corner during opponents turn. Edit: or maybe a timeglass, since the shield might make someone think that the defense value increases by defending like in later games.

A agree, that shield can confuse players from later HoMM games.
I also agree, that there's barely a space for 4 letters in that button. So, any Icon instead is a nice solution. Not sure about timeglass, though.

"Auto" for me looks quite universal. I believe this word could be understood by almost every player, no matter what is his native language.
For Cyrillic languages there's similar translation with same 4 symbols авто/ауто.

However, we can come up with common play icon or ⏯  for auto mode.

**Comment by Branikolog** [comment]
@zenseii 

> I want to add that if we make a skip button with an icon, then this icon would only be used when the resources are not English or Polish, and perhaps French. In all other cases we can safely replace the button.

That depends on the quality of buttons for these languages. While English has short word, that fits perfectly, Russian uses a shortened form of a longer word, which looks not really nice. So I'm OK with changing Russian skip button for a new icon.


>the hourglass icon might make players think we have implemented the wait function

100% agreed. 😅 However, right now I have no ideas in this regard. )

**Comment by felix-schneider** [comment]
My 2 cents: not many buttons in the Heroes 2 UI have icons (the only ones I can think of are the 8 buttons in the main adventure screen). I don't think an icon would fit with the rest of the UI.

I agree on "Auto" being universally understood. It would be up to the localizers to come up with a suitably short phrase to communicate the same if not.

**Comment by felix-schneider** [comment]
Fair point, I overlooked a couple. Still, I prefer the clarity of a text button. People can read different things into an icon (certainly I wouldn't associate a shield to mean "skip" or auto combat), the text can be more clear.

**Comment by felix-schneider** [comment]
The excessive hyphenation bothers me more than that. fheroes2 has an opportunity to fix that, though: make this button wider, to allow for localisations longer than four characters.

Not sure what I would go with in German... "Weiter"? "Nächste" (Kreatur)? "Warten"?

**Comment by felix-schneider** [comment]
This word exists in German, but is used almost exclusively in the context of card games.

Personally, I like "warten" (wait).

I only just now understood the context of the shield icon meaning "skip" in Heroes 3 (I was confused and thought it meant skip in a general UI context).

In Heroes 3, the shield in the combat screen meant defend, not skip. The creature skips it's turn in exchange for a defense boost of 20%. In that context, the shield makes sense. In Heroes 2, you get no sich bonus afaik, so the shield as a skip icon makes no sense.
